src/Compilers/Core/Portable/MetadataReader/PEModule.cs

```diff
@@ -38,7 +38,7 @@ internal sealed class PEModule : IDisposable
         private MetadataReader _lazyMetadataReader;
 
         private ImmutableArray<AssemblyIdentity> _lazyAssemblyReferences;
-        private Dictionary<string, AssemblyReferenceHandle> _lazyForwardedTypesToAssemblyMap;
+        private Dictionary<string, HashSet<AssemblyReferenceHandle>> _lazyForwardedTypesToAssemblyMap;
```

Comment:
Perhaps we should store a tuple instead of a HashSet, we probably are never interested in more than two references.